ui/src/tasks/actions/index.ts

```diff
@@ -460,10 +514,15 @@ export const getRuns = (taskID: string) => async (dispatch): Promise<void> => {
   try {
     dispatch(setRuns([], RemoteDataState.Loading))
 
-    const [runs] = await Promise.all([
-      client.tasks.getRunsByTaskID(taskID),
-      dispatch(selectTaskByID(taskID)),
-    ])
+    const resp = await getTasksRuns({taskID})
+
+    if (resp.status !== 200) {
+      throw new Error(resp.data.message)
+    }
+
+    let [runs] = await Promise.all([dispatch(selectTaskByID(taskID))])
+
+    runs = runs.concat(resp.data)
```

Comment:
selectTaskByID doesn't return anything, so runs is always undefined. and concat on undefined errors. 
try dispatch selectTaskByID above awaiting getTasksRuns. 
In this PR you are touching essential functionality of the UI, and nothing else matters if our data fetching is broken. Please test every place in the app that these functions are being used before moving on to refactoring the next function. The view task runs button on tasks does not work after this change. We don't have enough tests, and maybe that's the problem, but if we don't have tests, we can't move this fast.

Reply:
I truthfully hadn't even noticed that. Thanks for pointing that out!